File: src/main/java/com/gnefedev/gg/infrostructure/event/EventRegister.java
package com.gnefedev.gg.infrostructure.event;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

/**
 * Created by SBT-Nefedev-GV on 15.04.2016.
 */
@Component
public class EventRegister {
    private static ApplicationEventPublisher publisher;

    @Autowired
    public void setPublisher(ApplicationEventPublisher injected) {
        publisher = injected;
    }

    public static <T extends Event> void fire(T event) {
        publisher.publishEvent(event);
    }
}

File: src/main/java/com/gnefedev/gg/infrostructure/event/EventRegister.kt
package com.gnefedev.gg.infrostructure.event

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationEventPublisher
import org.springframework.context.ApplicationEventPublisherAware
import org.springframework.stereotype.Component

/**
 * Created by SBT-Nefedev-GV on 15.04.2016.
 */
@Component
object EventRegister : ApplicationEventPublisherAware {
    override fun setApplicationEventPublisher(publisher: ApplicationEventPublisher) {
        this.publisher = publisher;
    }
    private lateinit var publisher: ApplicationEventPublisher;

    @JvmStatic
    fun <T : Event> fire(event: T) {
        publisher.publishEvent(event)
    }
}
